marshal/marshal.go

```diff
@@ -82,13 +83,29 @@ func DecBigInt2C(data []byte, n *big.Int) {
 	}
 }
 
+// ReadBytes decodes bytes from the buffer.
+// It swallows whole p if there is not enough bytes for the data.
+// Deprecated: Use ReadBytes2 instead.
 func ReadBytes(p []byte) ([]byte, []byte) {
+	bytes, rest, err := ReadBytes2(p)
+	if err != nil {
+		return p, nil
+	}
+	return bytes, rest
+}
+
+// ReadBytes2 decodes bytes from the buffer.
+// It returns an error if there is not enough bytes to read the data.
+func ReadBytes2(p []byte) ([]byte, []byte, error) {
```

Comment:
named outputs could be really helpful for the users as there are two values of the same type. The same applies to the deprecated function too.

Something like:
```suggestion
func ReadBytes2(p []byte) (element, rest []byte, err error) {
```

Reply:
The function is used in generated code. Nevertheless someone might try using it manually, so I added the named return values. Thanks!